javascript/ql/src/semmle/javascript/security/dataflow/CleartextLoggingCustomizations.qll

```diff
@@ -32,6 +32,22 @@ module CleartextLogging {
    * A barrier for clear-text logging of sensitive information.
    */
   abstract class Barrier extends DataFlow::Node { }
+  
+  /**
+   * A call to `.replace()` that seems to mask 
+   */
+  class MaskingReplacer extends Barrier, DataFlow::MethodCallNode {
+  	MaskingReplacer() {
+  	  this.getCalleeName() = "replace" and
+  	  exists(RegExpLiteral reg| 
+  	    reg = this.getArgument(0).getALocalSource().asExpr() and
+  	    reg.getFlags().regexpMatch("(?i).*g.*") and
+  	    reg.getRoot().getRawValue().regexpMatch(".*\\..*")
+  	  ) 
+  	  and
+  	  this.getArgument(1).asExpr() instanceof StringLiteral
```

Comment:
- you can omit the `this.` prefix
- it is generally an antipattern to check for `StringLiteral`, consider using `ConstantString` instead to also handle the `TemplateLiteral` case. But in this case you can also just use `exists` and avoid the `asExpr()`.

Suggestion: 

 `exists(getArgument(1).getStringValue())`

(I assume you deliberately have omitted the potential dataflow step for the second argument.)

Reply:
> (I assume you deliberately have omitted the potential dataflow step for the second argument.)

Correct.